graphsdk/src/androidTest/java/com/microsoft/graph/concurrency/DefaultExecutorsTests.java

```diff
@@ -0,0 +1,133 @@
+package com.microsoft.graph.concurrency;
+
+import android.os.SystemClock;
+import android.test.AndroidTestCase;
+
+import com.microsoft.graph.core.ClientException;
+import com.microsoft.graph.core.GraphErrorCodes;
+import com.microsoft.graph.logger.MockLogger;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Test cases for {@see DefaultExecutors}
+ */
+public class DefaultExecutorsTests extends AndroidTestCase {
+
+    private MockLogger mLogger;
+    private DefaultExecutors defaultExecutors;
+    private String callbackResult;
+    private ClientException clientException;
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        mLogger = new MockLogger();
+        defaultExecutors = new DefaultExecutors(mLogger);
+    }
+
+    public void testNotNull() {
+        assertNotNull(mLogger);
+        assertNotNull(defaultExecutors);
+    }
+
+    public void testPerformOnBackground() {
+        String expectedLogMessage = "Starting background task, current active count: 0";
+        final AtomicBoolean success = new AtomicBoolean(false);
+        defaultExecutors.performOnBackground(new Runnable() {
+            @Override
+            public void run() {
+                success.set(true);
+            }
+        });
+        // sleep 1 second for background task
+        SystemClock.sleep(1000);
+        assertTrue(success.get());
+        assertEquals(1,mLogger.getLogMessages().size());
+        assertTrue(mLogger.isExist(expectedLogMessage));
+    }
+
+    public void testPerformOnForegroundWithResult() {
+        String expectedResult = "result value";
+        String expectedLogMessage = "Starting foreground task, current active count:0, with result result value";
+        final AtomicBoolean success = new AtomicBoolean(false);
+        final AtomicBoolean failure = new AtomicBoolean(false);
+        ICallback<String> callback = new ICallback<String>() {
+            @Override
+            public void success(String s) {
+                success.set(true);
+                callbackResult = s;
+            }
+
+            @Override
+            public void failure(ClientException ex) {
+                failure.set(true);
+            }
+        };
```

Comment:
Make a private class MockCallback() to reduce the duplicate code with this functionality.